cnd/src/network.rs

```diff
@@ -88,13 +91,33 @@ impl Swarm {
     pub async fn initiate_communication(
         &self,
         id: LocalSwapId,
-        peer: DialInformation,
+        DialInformation {
+            peer_id,
+            address_hint,
+        }: DialInformation,
```

Comment:
This syntax is super ugly. I get the feeling from the clunkyness of `DialInformation` that the fields are misnamed or the whole type is not useful.

Perhaps changing the `peer_id` field name to `id`
```

#[derive(Clone, Debug, PartialEq)]
pub struct DialInformation {
    pub id: PeerId,
    pub address_hint: Option<Multiaddr>,
}
```
This allows more ergonomic usage as such
```
fn foo(peer: DialInformation) {
    some_fn(peer.id)
    ...
}
```

Reply:
I think though that a better solution would be to remove `DialInformation` all together and use two arguments when they are needed.